ako-gateway-api/nodes/gateway_model_rel.go

```diff
@@ -588,3 +545,141 @@ func NoOperation(namespace, name, key string) ([]string, bool) {
+	for _, listener := range listeners {
+		//check if namespace is allowed
+		// TODO: akshay: add selector condition here.
+		if (len(listener.AllowedRouteTypes) == 0 || utils.HasElem(listener.AllowedRouteTypes, httpGroupKind)) &&
+			(listener.AllowedRouteNs == akogatewayapilib.AllowedRoutesNamespaceFromAll || listener.AllowedRouteNs == hrObj.Namespace) {
+			//if provided, check if section name and port matches
+			if (parentRef.SectionName == nil || string(*parentRef.SectionName) == listener.Name) &&
+				(parentRef.Port == nil || int32(*parentRef.Port) == listener.Port) {
+
+				gwListenerNsName := gwNsName + "/" + listener.Name
+				listenerHostname := akogatewayapiobjects.GatewayApiLister().GetGatewayListenerToHostname(gwListenerNsName)
+
+				hostnameMatched := false
+				for _, routeHostname := range hrObj.Spec.Hostnames {
+					// When Gateway hostname is empty, then just check validity of hostname and append it.
+					// When hostname in HTTProute has wildcard
+					// When there is exact match
+					if listenerHostname == "" || utils.CheckSubdomainOverlapping(string(routeHostname), listenerHostname) {
+						if akogatewayapilib.VerifyHostnameSubdomainMatch(string(routeHostname)) {
+							hostnameIntersection = append(hostnameIntersection, string(routeHostname))
+							hostnameMatched = true
+						}
+					}
+				}
+				if (hostnameMatched && !utils.HasElem(gatewayListenerList, listener)) || len(hrObj.Spec.Hostnames) == 0 {
+					gatewayListenerList = append(gatewayListenerList, listener)
+				}
+			}
+		}
+	}
+
+	if len(gatewayListenerList) > 0 {
+		gatewayToListenersMapList, _ := gatewayToListenersMap[gwNsName]
+		for _, gwListener := range gatewayListenerList {
+			if !utils.HasElem(gatewayToListenersMapList, gwListener) {
+				gatewayToListenersMapList = append(gatewayToListenersMapList, gwListener)
+			}
+		}
+		gatewayToListenersMap[gwNsName] = gatewayToListenersMapList
+	}
+	uniqueHosts := sets.NewString(hostnameIntersection...)
+	gwRouteNsName := fmt.Sprintf("%s/%s", gwNsName, routeTypeNsName)
+	akogatewayapiobjects.GatewayApiLister().UpdateGatewayRouteToHostname(gwRouteNsName, uniqueHosts.List())
+	akogatewayapiobjects.GatewayApiLister().UpdateGatewayRouteMappings(gwNsName, routeTypeNsName)
+	akogatewayapiobjects.GatewayApiLister().UpdateRouteToGatewayListenerMappings(gatewayToListenersMap[gwNsName], routeTypeNsName, gwNsName)
+	if !utils.HasElem(gwNsNameList, gwNsName) {
+		*gwNsNameList = append(*gwNsNameList, gwNsName)
+	}
+	parentNameToHostnameMap[string(parentRef.Name)] = hostnameIntersection
+}
+
+func validateReferredHTTPRoute(key, name, namespace string, allowedRoutesAll bool) ([]string, error) {
+	ns := namespace
+	if allowedRoutesAll {
+		ns = metav1.NamespaceAll
+	}
+	hrObjs, err := akogatewayapilib.AKOControlConfig().GatewayApiInformers().HTTPRouteInformer.Lister().HTTPRoutes(ns).List(labels.Set(nil).AsSelector())
+	httpRoutes := make([]*gatewayv1.HTTPRoute, 0)
```

Comment:
We can shift this line after `if err != nil` condition,

Reply:
Updated